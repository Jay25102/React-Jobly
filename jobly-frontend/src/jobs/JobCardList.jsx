import React from "react";
import JobCard from "./JobCard";

/**
 * renders JobCard components for each job passed into the function.
 */
function JobCardList({jobs}) {
    console.debug("JobCardList", "jobs=", jobs);

    return (
        <div className="JobCardList">
            {jobs.map(job => (
                <JobCard 
                    key={job.id}
                    id={job.id}
                    title={job.title}
                    salary={job.salary}
                    equity={job.equity}
                    companyName={job.companyName}
                />
            ))}
        </div>
    )
}

export default JobCardList;