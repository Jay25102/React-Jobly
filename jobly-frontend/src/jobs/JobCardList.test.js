import React from "react";
import { render } from "@testing-library/react";
import JobCardList from "./JobCardList";
import { UserProvider } from "../testUtils";

describe("JobCardList", function() {
    let jobs =  [{
        id: 1,
        title: "title",
        salary: 5555,
        equity: 5,
        companyName: "company name",
    }];
    test("it renders without crashing", function() {
        render(
            <UserProvider>
                <JobCardList jobs={jobs}/>                
            </UserProvider>
        );
    });
    test("it matches snapshot", function() {
        const { asFragment } = render(
            <UserProvider>
                <JobCardList jobs={jobs}/>                
            </UserProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
