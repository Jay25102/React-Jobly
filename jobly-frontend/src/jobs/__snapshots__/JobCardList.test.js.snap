// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCardList it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        Applied: 
      </div>
      <div>
        Title: title
      </div>
      <div>
        Company: company name
      </div>
      <div>
        Salary: 555,5
      </div>
      <div>
        Equity: 5
      </div>
      <button>
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
